pkg/cmd/cli/cmd/env.go

```diff
@@ -260,7 +265,19 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 				return nil
 			}
 			for _, c := range containers {
+				oldEnv := c.Env
+				for _, env := range remove {
+					_, ok := findEnv(c.Env, env)
+					if !ok {
+						infosMsgs[i] = append(infosMsgs[i], fmt.Sprintf("environment variable %q not found in %s container\n", env, c.Name))
+					}
+				}
 				c.Env = updateEnv(c.Env, env, remove)
+				if reflect.DeepEqual(oldEnv, c.Env) {
```

Comment:
This appears to set the updated flag based on the last container, so changes to earlier containers would be lost if the last container didn't change, right?